src/sqlfluff/core/config.py

```diff
@@ -195,26 +195,29 @@ class _RemovedConfig:
 ]
 
 
-def coerce_value(val: str) -> Any:
+def coerce_value(val: str) -> Union[int, float, bool, None, str]:
```

Comment:
Ha I almost prefer `Any` over a Union of 5 types, not sure

Reply:
Yeah that's fair - I'll revert this bit. It will result in the smallest PR ever though 🤣